herbie_core/services/message_publisher/message_publisher.py

```diff
@@ -0,0 +1,43 @@
+import logging
+from herbie_core.models import AbstractBusinessEntity
+from herbie_core.models.message_models_and_serializers import EntityUpdateMessage
+from herbie_core.models.message_models_and_serializers import EntityDeleteMessage
+from herbie_core.models.message_models_and_serializers import Message
+from herbie_core.services.message_publisher.registry import Registry
+from herbie_core.services.utils import BusinessEntityUtils
+
+
+class MessagePublisher:
+
+    def __init__(self):
+        self._logger = logging.getLogger(__name__)
+        self._publisher_list = Registry.get_publisher_list()
+
+    def send_entity_update_message(self, entity: AbstractBusinessEntity, tags=None):
+        if tags is None:
+            tags = []
+
+        update_message = EntityUpdateMessage(
+            BusinessEntityUtils.get_entity_type_name(entity),
+            entity.key,
+            entity.version,
+            entity.data,
+            entity.created,
+            entity.modified,
+            tags
+        )
+
+        self._send_message(update_message)
+
+    def send_entity_delete_message(self, entity: AbstractBusinessEntity):
+        delete_message = EntityDeleteMessage(
+            BusinessEntityUtils.get_entity_type_name(entity),
+            entity.key,
+            entity.version
+        )
+
+        self._send_message(delete_message)
+
+    def _send_message(self, message: Message):
+        for publisher in self._publisher_list:
```

Comment:
does this mean we will have 2 publishers here? LoggingPublisher and Kafka/GooglePubsub, and each message will be sent to 2 of them LoggingPublisher and Queue. In that case we don't use variable `self._logger` of the class? And is LoggingPublisher used only while sending message?

Reply:
Yes, my idea was that you can also have multiple publishers. One of the Publishers is just logging the message, which could be helpful for debugging.

Reply:
and in that case should that publisher be used in all modules for logging?